Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -610,6 +610,7 @@ internal void OnStopServer()
                 try
                 {
                     comp.OnStopServer();
+                    comp.ResetSyncObjects();
```

Comment:
once #1747  is merged,  this should be called inside the Reset method

Reply:
`Reset` is called before spawning, this will cause problems if items are added to a `SyncList` in `Awake` or `OnEnable`.

We need to reset the the list after `DestroyObject` is called (when scene objecta are hidden)

Reply:
after #1747 is merged,  Reset will be called when the object is destroyed,  not when it is spawned.